File: 02- February/21- Single Element in a Sorted Array/21- Single Element in a Sorted Array (Noura Algohary).py
# Author: Noura Algohary
class Solution:
    def singleNonDuplicate(self, nums: List[int]) -> int:
        low, high = 0, len(nums) - 1

        while(low < high):
            mid = int((low + high) / 2)

            if(mid % 2 != 0):
                mid -= 1
            
            # if the single element is on the left side 
            if nums[mid] == nums[mid + 1]:
                low = mid + 2
            else:
                high = mid
            
        return nums[high]

File: 02- February/21- Single Element in a Sorted Array/21- Single Element in a Sorted Array (Noura Algohary).cpp
// Author: Noura Algohary
class Solution {
public:
    int singleNonDuplicate(vector<int>& nums) {
        int low = 0, high = nums.size() - 1;

        while(low < high)
        {
            int mid = (low + high) / 2;

            if(mid % 2 !=0)
            {
                mid--;
            }
            // if the single element is on the left side 
            if(nums[mid] == nums[mid + 1])
            {
                low = mid + 2;
            }else
                high = mid;
        }
        return nums[high];
    }
};